tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -60,18 +60,55 @@ def handle_exceptions(self, exceptions):
         the stack trace"""
         raise exceptions[0]
 
-    def process_in_parallel(self, resources):
+    def process_in_parallel(self, resources, event):
         return ThreadHelper.execute_in_parallel(
             resources=resources,
-            execution_method=self.process_resource_set,
+            event=event,
+            execution_method=self._process_resources,
             executor_factory=self.executor_factory,
             log=self.log,
             max_workers=self.max_workers,
             chunk_size=self.chunk_size
         )
 
+    def _process_resources(self, resources, event):
+        self._prepare_processing()
+
+        for r in resources:
+            try:
+                self._process_resource(r)
+            except CloudError as e:
+                self.log.error("Failed to process resource.\n"
+                               "Type: {0}.\n"
+                               "Name: {1}.\n"
+                               "Error: {2}".format(r['type'], r['name'], e))
+
+    def _prepare_processing(self):
+        pass
+
+    @abc.abstractmethod
+    def _process_resource(self, resource):
+        raise NotImplementedError(
+            "Base action class does not implement behavior")
+
+
+@six.add_metaclass(abc.ABCMeta)
+class AzureEventAction(EventAction, AzureBaseAction):
+
+    def _process_resources(self, resources, event):
+        self._prepare_processing()
```

Comment:
Do you need to implement this? It should be available from AzureBaseAction already.